src/Bicep.LangServer/Completions/AutoCompleteFactory.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Navigation;
+using Bicep.Core.Semantics;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+
+namespace Bicep.LanguageServer.Completions
+{
+    /// <summary>
+    /// Class that constructs auto complete collections for the completion handler.
+    /// </summary>
+    internal class AutoCompleteFactory
+    {
+        private readonly CopilotManager copilotManager;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AutoCompleteFactory" /> class.
+        /// </summary>
+        public AutoCompleteFactory()
+        {
+            this.copilotManager = new CopilotManager();
+        }
+
+        /// <summary>
+        /// Factory method that builds out an implementation of the IAutoComplete interface based on whether or not it should resolve to a key or value autocomplete sequence.
+        /// </summary>
+        /// <param name="model">The SemanticModel.</param>
+        /// <param name="context">The BicepCompletionContext.</param>
+        /// <returns>Task that completes when the document autocomplete keywords are ready.</returns>
+        [SuppressMessage("Microsoft.Globalization", "CA1307:SpecifyStringComparison", Justification = "String literals are not locale-specific.")]
+        public async Task<IAutoComplete> BuildAsync(SemanticModel model, BicepCompletionContext context)
+        {
+            IAutoComplete copilotAutocompleteItems = await this.AutocompleteRuleConditionAsync(model, context).ConfigureAwait(false);
```

Comment:
The name of this function is not in sync with its behavior. Consider that we are not completing a `Policy Rule Condition`, rather we are doing a code completion for a Bicep resource type body